client/my-sites/themes/controller.jsx

```diff
@@ -77,19 +91,51 @@ export function loggedOut( context, next ) {
 	next();
 }
 
-export function fetchThemeData( context, next ) {
+export function fetchThemeData( context, next, shouldUseCache = false ) {
 	if ( ! context.isServerSide ) {
 		return next();
 	}
 
 	const queryParams = {
 		search: context.query.s,
 		tier: context.params.tier,
-		filter: context.params.filter,
+		filter: compact( [ context.params.filter, context.params.vertical ] ).join( ',' ),
 		page: 0,
 		perPage: PER_PAGE,
 	};
+	const cacheKey = context.path;
 
 	context.store.dispatch( query( queryParams ) );
-	context.store.dispatch( fetchNextPage( false ) ).then( () => next() );
+	context.store.dispatch( incrementThemesPage( false ) );
+
+	if ( shouldUseCache ) {
+		const cachedData = themesQueryCache.get( cacheKey );
+		if ( cachedData ) {
+			debug( `found theme data in cache key=${ cacheKey }` );
+			context.store.dispatch( cachedData.action );
+			context.renderCacheKey = context.path + cachedData.timestamp;
+			return next();
+		}
+	}
+
+	context.store.dispatch( fetchThemes( false ) )
```

Comment:
I agree. For reasons (described below) this parameter must be a site object or `false`, so I suppose the quick fix is just to set a default params in the action creator definition (`function fetchThemes( site = false )`)?

For context, [`fetchThemes` passes its `site` param to `wpcom.undocumented.themes()`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/wpcom-undocumented/lib/undocumented.js#L1458), which [asks for properties like `site.ID`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/wpcom-undocumented/lib/undocumented.js#L1458). If we pass nothing, `site` will be `undefined`, and asking for properties of `undefined` leads to runtime errors. However because JavaScript is magical and inscrutable, passing `false` works because it is object-like and asking for missing properties will simply turn up `undefined`. See:

<img width="475" alt="screen shot 2016-11-04 at 6 37 19 am" src="https://cloud.githubusercontent.com/assets/518059/20004250/3092fc20-a259-11e6-9c81-e817ae218e36.png">

This seems to be the expected behavior for passing a `site` around too, such as `ThemesListFetcher` which [requires `site` to be an object or a boolean](https://github.com/Automattic/wp-calypso/blob/master/client/components/data/themes-list-fetcher/index.jsx#L24-L27) and [defaults `site` to false](https://github.com/Automattic/wp-calypso/blob/master/client/components/data/themes-list-fetcher/index.jsx#L102).

I'm happy with a default param solution to make this less confusing — but just noting that there's a lot of other action creators that rely on `site`, so should this be a universal solution everywhere?